android/src/main/java/com/tailscale/ipn/AppSourceChecker.kt

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+package com.tailscale.ipn
+
+import android.content.Context
+import android.os.Build
+import android.util.Log
+
+object AppSourceChecker {
+
+  const val TAG = "AppSourceChecker"
+
+  fun isInstalledFromGooglePlay(context: Context): Boolean {
```

Comment:
Could we cache the resulting value of this function in a variable somewhere, so we don't run this logic multiple times per app execution? Calls into PackageManager are slow, and the result is guaranteed to not change across calls here.

Reply:
It's called once by runBackend

Reply:
👍🏻

Reply:
I was concerned about the backend calling this repeatedly in the background every time hostinfo gets created, but this is good.